functions

```diff
@@ -21,87 +21,125 @@ function resultCheck() {
 
 # Takes an expected string and returned value
 # Usage: resultCheckString "This is the title" $RETURNED_TITLE
+# Deprecated: This is the same as resultCheck, just whether you wrap the 
+#   calling arguments in quotes.
+# ie. resultCheck 200 $Some_integer_or_single_word
+#     resultCheck "Some text to match" "$A_variable_with_text_containing_spaces"
 function resultCheckString() {
-  EXPECTED=$1
-  ACTUAL=$2
-  if [ "$EXPECTED" != "$ACTUAL" ]; then
-    echo "ERROR expected $EXPECTED and received $ACTUAL"
-    exit 1
-  fi
-  echo "$EXPECTED == $ACTUAL (Pass)"
-  return 0
+  local EXPECTED=$1
+  local ACTUAL=$2
+  resultCheck $1 $2
 }
 
 # Takes an expected Status Code and grabs the other from Headers.
 # NOTE: Header variable must be wrapped in quotes when calling.
 # Usage: resultCheckInHeaders 200 "$RETURNED_CODE"
 function resultCheckInHeaders() {
-  EXPECTED=$1
-  HEADERS=$2
-  ACTUAL=$(echo "$HEADERS" | grep 'HTTP/1.1' | cut -d' ' -f2)
-  if [ "$EXPECTED" -ne "$ACTUAL" ]; then
-    echo "ERROR expected $EXPECTED and received $ACTUAL"
-    if [ "$ACTUAL" == "405" ]; then
-      echo "ERROR check the FEDORA_PATH setting in the config file to ensure you're pointing to the repository correctly."
+  local EXPECTED=$1
+  local HEADERS=$2
+  for ACTUAL in $(echo "$HEADERS" | grep 'HTTP/1.1' | cut -d' ' -f2); do
+    if [ "$EXPECTED" == "$ACTUAL" ]; then
+      echo "$EXPECTED == $ACTUAL (Pass)"
+      return 0
+    elif [ "100" == "$ACTUAL" ]; then
+        continue
+    else
+      echo "ERROR expected $EXPECTED and received $ACTUAL"
+      if [ "$ACTUAL" == "405" ]; then
+        echo "ERROR check the FEDORA_PATH setting in the config file to ensure you're pointing to the repository correctly."
+      fi
+      exit 1
     fi
-
-    exit 1
-  fi
-  echo "$EXPECTED == $ACTUAL (Pass)"
-  return 0
+  done
+  echo "Unable to locate HTTP/1.1 header"
+  exit 1
 }
 
 # Find the Location: header and return the value in the LOCATION variable.
 # NOTE: Header variable must be wrapped in quotes when calling.
 # Usage: getLocationFromHeaders "$HEADERS"
 # Return: Location URL in the $LOCATION variable.
 function getLocationFromHeaders() {
-  HEADERS=$1
-  export LOCATION=$(echo "$HEADERS" | grep '^Location:' | cut -d' ' -f2 | tr -d '\r')
+  local HEADERS=$1
+  local LOCATION=$(echo "$HEADERS" | grep '^Location:' | cut -d' ' -f2 | tr -d '\r')
```

Comment:
In order for the `LOCATION` variable to be available in other scripts, this should be `export` instead of `local`.
See failure of `./transaction_tests.sh`

Reply:
If it finds the location header, it `echo`s it out do avoid `export`ing. This allows you to store the LOCATION header from multiple requests as it is not re-using a variable. Instead you call this function like
```
LOCATION=$(getLocationFromHeaders "$HTTP_RES")
```
or
```
my_new_location_variable=$(getLocationFromHeaders "$HTTP_RES")
```

I thought I had caught all the places to accommodate this change in usage.

Reply:
I am seeing the `TRANSACTION` variable as being empty when I run `transaction_tests.sh`.
https://github.com/whikloj/fcrepo4-tests/blob/master/transaction_tests.sh#L18

Reply:
Oh right, I assumed transactions were dead until the sidecar spec landed. I should have removed that test file.

Reply:
I think as long as transactions are in the release, we should make sure they work. Your test script helps.